src/mscorlib/Resources/Strings.resx

```diff
@@ -1577,7 +1577,7 @@
     <value>The unmanaged Version information is too large to persist.</value>
   </data>
   <data name="Argument_WaitHandleNameTooLong" xml:space="preserve">
-    <value>The name can be no more than {0} characters in length.</value>
+    <value>The name can be no more than {0} characters in length: {1}.</value>
```

Comment:
To be consistent with path errors, I suggest

`The name '{0}' can be no more than '{1}' characters long.`